src/foam/java/refinements.js

```diff
@@ -212,11 +214,28 @@ foam.CLASS({
     },
     {
       class: 'String',
-      name: 'javaValidateObj'
+      name: 'javaValidateObj',
+      expression: function(validationPredicates) {
+        return validationPredicates
+          .map(vp => {
+            return `
+if ( ! ${foam.java.asJavaValue(vp.predicate)}.f(obj) ) {
+  throw new IllegalStateException(${foam.java.asJavaValue(vp.errorString)});
```

Comment:
I think we need to model a specific ValidationException that the JS side can catch and handle specially.

Reply:
Agreed. We must do something already because I've seen some server side errors bubble up to the client but I'm not entirely sure how that's happening.

Reply:
WizardView in particular may want to handle ValidationExpression well, as it could jump to the appropriate section if it wanted.